packages/cli-kit/src/environment/local.test.ts

```diff
@@ -128,3 +128,60 @@ describe('analitycsDisabled', () => {
     expect(got).toBe(false)
   })
 })
+
+describe('useDeviceAuth', () => {
+  it('returns true when SHOPIFY_CLI_DEVICE_AUTH is truthy', () => {
+    // Given
+    const env = {SHOPIFY_CLI_DEVICE_AUTH: '1'}
+
+    // When
+    const got = useDeviceAuth(env)
+
+    // Then
+    expect(got).toBe(true)
+  })
+
+  it('returns true when SPIN is truthy', () => {
+    // Given
+    const env = {SPIN: '1'}
+
+    // When
+    const got = useDeviceAuth(env)
+
+    // Then
+    expect(got).toBe(true)
+  })
+
+  it('returns true when CODESPACES is truthy', () => {
+    // Given
+    const env = {CODESPACES: '1'}
+
+    // When
+    const got = useDeviceAuth(env)
+
+    // Then
+    expect(got).toBe(true)
+  })
+
+  it('returns true when GITPOD_WORKSPACE_ID is truthy', () => {
+    // Given
+    const env = {GITPOD_WORKSPACE_ID: '1'}
+
+    // When
+    const got = useDeviceAuth(env)
+
+    // Then
+    expect(got).toBe(true)
+  })
+
+  it('returns false otherwise', () => {
```

Comment:
This `otherwise` makes assumption regarding the position of the test inside the `describe` block. I'd consider changing it to: `returns false when GITPOD_WORKSPACE_ID if falsy`.